api/v1alpha1/observatorium_types.go

```diff
@@ -0,0 +1,143 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ObservatoriumSpec defines the desired state of Observatorium
+type ObservatoriumSpec struct {
+	// Thanos Image name
+	Image string `json:"thanosImage"`
+	// Thanos Image version
+	Version string `json:"thanosVersion"`
+	// Objest Storage Configuration
+	ObjectStorageConfig ObjectStorageConfig `json:"objectStorageConfig"`
+	// Hashrings describes a list of Hashrings
+	Hashrings []*Hashring `json:"hashrings"`
+	// Thanos CompactSpec
+	Compact CompactSpec `json:"compact"`
+	// Thanos Receive Controller Spec
+	ReceiveController ReceiveControllerSpec `json:"thanosReceiveController"`
+	// Thanos ThanosPersistentSpec
+	Receivers ReceiversSpec `json:"receivers"`
+	// Thanos QuerierCache
+	QueryCache QuerierCacheSpec `json:"queryCache"`
+	// Thanos StoreSpec
+	Store StoreSpec `json:"store"`
+	// Thanos RulerSpec
+	Rule RuleSpec `json:"rule"`
+	// API Gateway
+	ApiGateway ApiGatewaySpec `json:"apiGateway"`
+}
+
+type ObjectStorageConfig struct {
+	// Object Store Config Secret Name for Thanos
+	Name string `json:"name"`
+	// Object Store Config key for Thanos
+	Key string `json:"key"`
+}
+
+type ReceiveControllerSpec struct {
+	Image string `json:"image"`
+	// Tag describes the tag of Thanos receive controller to use.
+	Version string `json:"version,omitempty"`
+}
+
+type ReceiversSpec struct {
+	// VolumeClaimTemplate
+	VolumeClaimTemplate VolumeClaimTemplate `json:"volumeClaimTemplate"`
+}
+
+type StoreSpec struct {
+	// VolumeClaimTemplate
+	VolumeClaimTemplate VolumeClaimTemplate `json:"volumeClaimTemplate"`
+	Shards              *int32              `json:"shards,omitempty"`
+}
+
+type ApiGatewaySpec struct {
+	Image string `json:"image"`
+	// Tag describes the tag of Thanos receive controller to use.
+	Version string `json:"version,omitempty"`
+}
+
+type RuleSpec struct {
+	// VolumeClaimTemplate
+	VolumeClaimTemplate VolumeClaimTemplate `json:"volumeClaimTemplate"`
+}
+
+type CompactSpec struct {
+	// VolumeClaimTemplate
+	VolumeClaimTemplate VolumeClaimTemplate `json:"volumeClaimTemplate"`
+	// RetentionResolutionRaw
+	RetentionResolutionRaw string `json:"retentionResolutionRaw"`
+	// RetentionResolutionRaw
+	RetentionResolution5m string `json:"retentionResolution5m"`
+	// RetentionResolutionRaw
+	RetentionResolution1h string `json:"retentionResolution1h"`
+}
+
+type VolumeClaimTemplate struct {
+	VolumeClaimTemplateSpec v1.PersistentVolumeClaimSpec `json:"spec"`
+}
+
+type QuerierCacheSpec struct {
+	// Thanos Querier Cache  name
```

Comment:
Is this really the name? Or is this the image?

Reply:
Done